### PR TITLE
add german translation to client and server

### PR DESCRIPTION
**Pull request by quasigroup**
Add german translation (by Eudoxia and Quasigroup) to the client and the server part.

### CODE REVIEW

**Comment by jdmonin** [comment]
Thank you very much, this is a nice contribution! Doubly so since the board game originated in Germany.

I'd like to give you credit for it in the release notes ([Versions.md](https://github.com/jdmonin/JSettlers2/blob/main/doc/Versions.md)). For example, v2.5.00 has
```
- Added French translation (thank you Lee Passey)
```

A psydonym or username (like "Eudoxia and Quasigroup") is fine for that if you'd prefer.

Could you please update the "This file Copyright" lines in your `.properties` files to give yourself attribution?

File: src/main/resources/resources/strings/client/data_de.properties [hunk 1]
```diff
@@ -0,0 +1,806 @@
+# JSettlers i18n localization strings: At Client.
+#  This file is part of Java Settlers - An online multiplayer version of the game Settlers of Catan
+#
+# This file Copyright (C) 2013 Luis A. Ramirez <lartkma@gmail.com>
```
Comment (jdmonin) [inline src/main/resources/resources/strings/client/data_de.properties:4]: Can you please update this line to give yourself a quick attribution? An alias or nickname is fine if you'd prefer that. The email part is also optional. Year should be 2023 :)

File: src/main/resources/resources/strings/server/toClient_de.properties [hunk 2]
```diff
@@ -0,0 +1,455 @@
+# JSettlers i18n localization strings: At Server to send to Client.
+#  This file is part of Java Settlers - An online multiplayer version of the game Settlers of Catan
+#
+# This file Copyright (C) 2013-2023 Jeremy D Monin <jeremy@nand.net>
```
Comment (jdmonin) [inline src/main/resources/resources/strings/server/toClient_de.properties:4]: This can be a copy-paste of the attribution line in `data_de.properties`

**Comment by quasigroup** [comment]
> Thank you very much, this is a nice contribution! Doubly so since the board game originated in Germany.
> 
> I'd like to give you credit for it in the release notes ([Versions.md](https://github.com/jdmonin/JSettlers2/blob/main/doc/Versions.md)). For example, v2.5.00 has
> 
> ```
> - Added French translation (thank you Lee Passey)
> ```
> 
> A psydonym or username (like "Eudoxia and Quasigroup") is fine for that if you'd prefer.
> 
> Could you please update the "This file Copyright" lines in your `.properties` files to give yourself attribution?

I've updates the properties file and would be glad if you would mention our pseudonyms in the release notes.

**Review by jdmonin — APPROVED** [review]

**Comment by jdmonin** [comment]
Great, thanks again!

**Pull request merged**